irohad/ametsuchi/impl/rocksdb_block_storage.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/rocksdb_block_storage.hpp"
+
+#include "ametsuchi/impl/rocksdb_common.hpp"
+#include "backend/protobuf/block.hpp"
+#include "common/byteutils.hpp"
+#include "logger/logger.hpp"
+
+using namespace iroha::ametsuchi;
+
+RocksDbBlockStorage::RocksDbBlockStorage(
+    std::shared_ptr<RocksDBContext> db_context,
+    std::shared_ptr<shared_model::interface::BlockJsonConverter> json_converter,
+    logger::LoggerPtr log)
+    : db_context_(std::move(db_context)),
+      json_converter_(std::move(json_converter)),
+      log_(std::move(log)) {}
+
+bool RocksDbBlockStorage::insert(
+    std::shared_ptr<const shared_model::interface::Block> block) {
+  return json_converter_->serialize(*block).match(
+      [&](const auto &block_json) {
```

Comment:
could you please remake this spagetti. name this lambda as `handle_serialized_block()`, and the second one `handle_error()`